app/src/main/java/com/example/plantsapp/data/contentprovider/PlantsContentProvider.kt

```diff
@@ -47,14 +57,32 @@ class PlantsContentProvider : ContentProvider() {
         selectionArgs: Array<out String>?,
         sortOrder: String?
     ): Cursor {
-        return when (uri.toString()) {
-            PlantStatisticsContract.CONTENT_URI -> createPlantsCursor()
+        return when (uriMatcher.match(uri)) {
+            PLANTS_URI_CODE -> runBlocking { createPlantsCursorUseCase() }
+            TASKS_URI_CODE -> runBlocking {
+                createTasksCursorUseCase(
+                    plantName = selectionArgs?.first()
+                        ?: throw IllegalArgumentException("Selection argument is required for this query")
+                )
+            }
+            TASK_HISTORY_URI_CODE -> runBlocking {
+                createTaskHistoryCursorUseCase(
+                    plantName = selectionArgs?.first()
+                        ?: throw IllegalArgumentException("Selection arguments are required for this query"),
+                    taskKey = selectionArgs[1]
+                )
+            }
```

Comment:
How about hide it into suspend private function and just call runBlocking only once when you call this method